src/mechanics/mesh/MeshGmsh.cpp

```diff
@@ -0,0 +1,375 @@
+#include "MeshGmsh.h"
+
+#include "base/Exception.h"
+#include "mechanics/interpolation/InterpolationTrussLinear.h"
+#include "mechanics/interpolation/InterpolationTriangleLinear.h"
+#include "mechanics/interpolation/InterpolationQuadLinear.h"
+
+#include <array>
+#include <climits>
+#include <algorithm>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
+
+
+// Helper structs (cpp only)
+// %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+struct GmshHeader
+{
+    double version;
+    bool isBinary;
+    int doubleSize;
+};
+
+struct GmshNode
+{
+    int id = 0;
+    std::array<double, 3> coordinates;
+};
+
+struct GmshElement
+{
+    int id = 0;
+    int type = 0;
+    std::vector<int> tags;
+    std::vector<int> nodes;
+};
+
+struct GmshPhysicalNames
+{
+    int id = 0;
+    int dimension = 0;
+    std::string name;
+};
+
+
+struct GmshFileContent
+{
+    GmshHeader header;
+    int minNodeId = INT_MAX;
+    int maxNodeId = INT_MIN;
+    int dimension = 1;
+    std::vector<GmshNode> nodes;
+    std::vector<GmshElement> elements;
+    std::vector<GmshPhysicalNames> physicalNames;
+};
+
+// Helper functions (cpp only)
+// %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+static GmshHeader ReadGmshHeader(std::ifstream& file)
```

Comment:
What does this `static` do?

Reply:
whoops that should be gone... I ll remove that